contributors/design-proposals/raw-block-pv.md

```diff
@@ -0,0 +1,520 @@
+# Raw Block Consumption in Kubernetes
+
+Authors: erinboyd@, screeley44@, mtanino@
+
+This document presents a proposal for managing raw block storage in Kubernetes using the persistent volume source API as a consistent model of consumption.
+
+# Terminology
+* Raw Block Device - a physically attached device devoid of a filesystem
+* Raw Block Volume - a logical abstraction of the raw block device as defined by a path
+* File on Block/Filesystem on Block - a formatted (ie xfs) filesystem on top of a raw block device
+
+# Goals
+* Enable durable access to block storage
+* Support storage requirements for all workloads supported by Kubernetes
+* Provide flexibility for users/vendors to utilize various types of storage devices
+* Agree on API changes for block
+* Provide a consistent security model for block devices 
+* Provide a means for running containerized block storage offerings as non-privileged container
+
+# Non Goals
+* Support all storage devices natively in upstream Kubernetes. Non-standard storage devices are expected to be managed using extension
+  mechanisms.
+   
+# Value add to Kubernetes
+
+  Before the advent of storage plugins, emptyDir and hostPath were widely used to quickly prototype stateless applications in Kubernetes. 
+  Both have limitations for their use in application that need to store persistent data or state. 
+  EmptyDir, though quick and easy to use, provides no real guarantee of persistence for suitable amount of time. 
+  Appropriately used as scratch space, it does not have the HostPath, became an initial offering for local storage, but had many
+  drawbacks. Without having the ability to guarantee space & ensure ownership, one would lose data once a node was rescheduled. 
+  Therefore, the risk outweighed the reward when trying to  leverage the power of local storage needed for stateful applications like 
+  databases.
+  
+  By extending the API for volumes to specifically request a raw block device, we provide an explicit method for volume comsumption,
+  whereas previously it was always a fileystem. In addition, the ability to use a raw block device without a filesystem will allow
+  Kubernetes better support of high performance applications that can utilize raw block devices directly for their storage. 
+  For example, MariaDB or MongoDB.
+  
+  Specific use cases around improved usage of storage consumption are included in the use cases listed below as follows:
```

Comment:
The use cases listed here are better articulated in the 'Use Cases' section below.  It may be most useful to elevate the high level use cases as:
- Database applications that can read/write directly to block storage
- Storage platforms that use block storage as the backing storage
- FileSystem utilities, monitoring & host bootstrapping which would benefit from direct access to block storage.